### PR TITLE
Update AnimatePacket.php

### PR DESCRIPTION
**Pull request by NycuRO**

### CODE REVIEW

**Comment by NycuRO** [comment]
@armagadon159753

**Comment by armagadon159753** [comment]
^~^ i love you

File: src/pocketmine/network/protocol/AnimatePacket.php [hunk 1]
```diff
@@ -36,8 +36,8 @@ class AnimatePacket extends DataPacket {
 	 *
 	 */
 	public function decode(){
-		$this->action = $this->getVarInt();
-		$this->eid = $this->getEntityId();
+		$this->action = (int) $this->getUnsignedVarInt();
+		$this->eid = $this->getVarLong();
```
Comment (SuperXingKong) [inline src/pocketmine/network/protocol/AnimatePacket.php]: Incorret here...

**Comment by armagadon159753** [comment]
#677

**Comment by SuperXingKong** [comment]
@armagadon159753 How to reproduce it? I didn't have this error.

**Comment by armagadon159753** [comment]
seriously?!
You have to get on a boat
@SuperXingKong

**Comment by SuperXingKong** [comment]
@armagadon159753 Yes, I am rowing a boat but no errors.

File: src/pocketmine/network/protocol/AnimatePacket.php [hunk 2]
```diff
@@ -30,14 +30,14 @@ class AnimatePacket extends DataPacket {
 
 	public $action;
 	public $eid;
-	public $float;
+	public $float = 0.0; //TODO (Boat rowing time)
```
Comment (NycuRO) [inline src/pocketmine/network/protocol/AnimatePacket.php]: :)))))))))

File: src/pocketmine/network/protocol/AnimatePacket.php (hunk 1 above)
Comment (NycuRO) [inline src/pocketmine/network/protocol/AnimatePacket.php]: @SuperXingKong Are you sure? 👍

**Comment by DerddyDert** [comment]
@SuperXingKong wait for me on that boat, lets go fishing to catch dinner :-p

File: src/pocketmine/network/protocol/AnimatePacket.php (hunk 1 above)
Comment (SuperXingKong) [inline src/pocketmine/network/protocol/AnimatePacket.php]: @NycuRO I‘m sure. Why did you use getVarLong to  get Entityid?

**Comment by SuperXingKong** [comment]
@DerddyDert Good luck lol

**Comment by DerddyDert** [comment]
@NycuRO @SuperXingKong where are we with this PR?

**Comment by NycuRO** [comment]
??

**Comment by DerddyDert** [comment]
@NycuRO He made recommendations. Is this ready to merge?

**Comment by NycuRO** [comment]
Yes.

**Pull request merged**